Objects/stringlib/unicode_format.h

```diff
@@ -462,8 +462,20 @@ get_field_object(SubString *input, PyObject *args, PyObject *kwargs,
             tmp = getattr(obj, &name);
         else
             /* getitem lookup "[]" */
-            if (index == -1)
+            if (index == -1) {
+                if (PySequence_Check(obj)) {
+                    /* string index can't be passed to sequence */
+                    PyObject *str = SubString_new_object(&name);
+                    if (str != NULL)
```

Comment:
```suggestion
                    if (str)
```

Also beware PEP 7: "braces are required everywhere, even where C permits them to be omitted".